backend/src/main/java/de/ffmjava/capstone/backend/stock/StockService.java

```diff
@@ -16,6 +18,13 @@ public List<StockItem> getAllStockItems() {
         return repository.findAll();
     }
 
+    public void deleteStockItem(String id) throws ResponseStatusException{
+        if (!repository.existsById(id)) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Kein Eintrag für die gegebene ID gefunden");
```

Comment:
It gets re-thrown in the controller, I like to keep the controller relatively clean of logic. Also feels cleaner since the repository is not needed in the controller as a dependency and therefore the only class it can communicate with is the service